File: WidgetDesigners/Authorization/AuthorizationDesigner.js
Type.registerNamespace("SitefinityWebApp.WidgetDesigners.Authorization");

SitefinityWebApp.WidgetDesigners.Authorization.AuthorizationDesigner = function (element) {
    /* Initialize AuthorizationUrl fields */
    this._authorizationUrl = null;
    
    /* Initialize UnauthorizationUrl fields */
    this._unauthorizationUrl = null;
    
    /* Calls the base constructor */
    SitefinityWebApp.WidgetDesigners.Authorization.AuthorizationDesigner.initializeBase(this, [element]);
}

SitefinityWebApp.WidgetDesigners.Authorization.AuthorizationDesigner.prototype = {
    /* --------------------------------- set up and tear down --------------------------------- */
    initialize: function () {
        /* Here you can attach to events or do other initialization */
        SitefinityWebApp.WidgetDesigners.Authorization.AuthorizationDesigner.callBaseMethod(this, 'initialize');
    },
    dispose: function () {
        /* this is the place to unbind/dispose the event handlers created in the initialize method */
        SitefinityWebApp.WidgetDesigners.Authorization.AuthorizationDesigner.callBaseMethod(this, 'dispose');
    },

    /* --------------------------------- public methods ---------------------------------- */

    findElement: function (id) {
        var result = jQuery(this.get_element()).find("#" + id).get(0);
        return result;
    },

    /* Called when the designer window gets opened and here is place to "bind" your designer to the control properties */
    refreshUI: function () {
        var controlData = this._propertyEditor.get_control().Settings; /* JavaScript clone of your control - all the control properties will be properties of the controlData too */

        /* RefreshUI AuthorizationUrl */
        jQuery(this.get_authorizationUrl()).val(controlData.AuthorizationUrl);

        /* RefreshUI UnauthorizationUrl */
        jQuery(this.get_unauthorizationUrl()).val(controlData.UnauthorizationUrl);
    },

    /* Called when the "Save" button is clicked. Here you can transfer the settings from the designer to the control */
    applyChanges: function () {
        var controlData = this._propertyEditor.get_control().Settings;

        /* ApplyChanges AuthorizationUrl */
        controlData.AuthorizationUrl = jQuery(this.get_authorizationUrl()).val();

        /* ApplyChanges UnauthorizationUrl */
        controlData.UnauthorizationUrl = jQuery(this.get_unauthorizationUrl()).val();
    },

    /* --------------------------------- event handlers ---------------------------------- */

    /* --------------------------------- private methods --------------------------------- */

    /* --------------------------------- properties -------------------------------------- */

    /* AuthorizationUrl properties */
    get_authorizationUrl: function () { return this._authorizationUrl; }, 
    set_authorizationUrl: function (value) { this._authorizationUrl = value; },

    /* UnauthorizationUrl properties */
    get_unauthorizationUrl: function () { return this._unauthorizationUrl; }, 
    set_unauthorizationUrl: function (value) { this._unauthorizationUrl = value; }
}

SitefinityWebApp.WidgetDesigners.Authorization.AuthorizationDesigner.registerClass('SitefinityWebApp.WidgetDesigners.Authorization.AuthorizationDesigner', Telerik.Sitefinity.Web.UI.ControlDesign.ControlDesignerBase);

File: WidgetDesigners/Authorization/AuthorizationDesigner.cs
using System;
using System.Linq;
using System.Web.UI;
using Telerik.Sitefinity.Web.UI;
using Telerik.Sitefinity.Web.UI.ControlDesign;
using System.Collections.Generic;

namespace SitefinityWebApp.WidgetDesigners.Authorization
{
    /// <summary>
    /// Represents a designer for the <typeparamref name="SitefinityWebApp.Mvc.Controllers.AuthorizationController"/> widget
    /// </summary>
    public class AuthorizationDesigner : ControlDesignerBase
    {
        #region Properties
        /// <summary>
        /// Obsolete. Use LayoutTemplatePath instead.
        /// </summary>
        protected override string LayoutTemplateName
        {
            get
            {
                return string.Empty;
            }
        }

        /// <summary>
        /// Gets the layout template's relative or virtual path.
        /// </summary>
        public override string LayoutTemplatePath
        {
            get
            {
                if (string.IsNullOrEmpty(base.LayoutTemplatePath))
                    return AuthorizationDesigner.layoutTemplatePath;
                return base.LayoutTemplatePath;
            }
            set
            {
                base.LayoutTemplatePath = value;
            }
        }

        protected override HtmlTextWriterTag TagKey
        {
            get
            {
                return HtmlTextWriterTag.Div;
            }
        }
        #endregion

        #region Control references
        /// <summary>
        /// Gets the control that is bound to the AuthorizationUrl property
        /// </summary>
        protected virtual Control AuthorizationUrl
        {
            get
            {
                return this.Container.GetControl<Control>("AuthorizationUrl", true);
            }
        }

        /// <summary>
        /// Gets the control that is bound to the UnauthorizationUrl property
        /// </summary>
        protected virtual Control UnauthorizationUrl
        {
            get
            {
                return this.Container.GetControl<Control>("UnauthorizationUrl", true);
            }
        }

        #endregion

        #region Methods
        protected override void InitializeControls(Telerik.Sitefinity.Web.UI.GenericContainer container)
        {
            // Place your initialization logic here
        }
        #endregion

        #region IScriptControl implementation
        /// <summary>
        /// Gets a collection of script descriptors that represent ECMAScript (JavaScript) client components.
        /// </summary>
        public override System.Collections.Generic.IEnumerable<System.Web.UI.ScriptDescriptor> GetScriptDescriptors()
        {
            var scriptDescriptors = new List<ScriptDescriptor>(base.GetScriptDescriptors());
            var descriptor = (ScriptControlDescriptor)scriptDescriptors.Last();

            descriptor.AddElementProperty("authorizationUrl", this.AuthorizationUrl.ClientID);
            descriptor.AddElementProperty("unauthorizationUrl", this.UnauthorizationUrl.ClientID);

            return scriptDescriptors;
        }

        /// <summary>
        /// Gets a collection of ScriptReference objects that define script resources that the control requires.
        /// </summary>
        public override System.Collections.Generic.IEnumerable<System.Web.UI.ScriptReference> GetScriptReferences()
        {
            var scripts = new List<ScriptReference>(base.GetScriptReferences());
            scripts.Add(new ScriptReference(AuthorizationDesigner.scriptReference));
            return scripts;
        }
        #endregion

        #region Private members & constants
        public static readonly string layoutTemplatePath = "~/WidgetDesigners/Authorization/AuthorizationDesigner.ascx";
        public const string scriptReference = "~/WidgetDesigners/Authorization/AuthorizationDesigner.js";
        #endregion
    }
}
 
